File: drish/cs50/cash.py
from cs50 import get_float


def main():
    n = coinage(0.25, 0.10, 0.05, 0.01)
    print(n)


def coinage(*denom):
    n = 0
    owed = round(100 * get_positive_float("Change owed: "))  # prompt user for input # round to avoud floating point imprecision
    for i in range(len(denom)):
        while owed >= round(100 * denom[i]):
            n += 1
            owed -= round(100 * denom[i])
    return n


def get_positive_float(prompt):  # define function to only accept positive float input
    while True:
        n = get_float(prompt)
        if n > 0:
            break
    return n


main()

File: drish/cs50/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>


int main(void)
{
    while(true) //set up a while loop so invalid inputs re-prompt the user
    { float rawfax = get_float("Change owed: ");
        if (rawfax > 0)
        {
            int counter = 0;
            int centsowed = round(rawfax * 100);
                for (; centsowed >= 25; counter++)
                        {
                            centsowed = centsowed - 25;             
                        }
                for (; centsowed >= 10; counter++)
                        {
                            centsowed = centsowed - 10;             
                        }
                for (; centsowed >= 5; counter++)
                        {
                            centsowed = centsowed - 5;             
                        }
                for (; centsowed >= 1; counter++)
                        {
                            centsowed = centsowed - 1;             
                        }
            printf("%i\n", counter);
            break;
                }        
        }
}
